volumes/ecocounter/pull-data-from-api.py

```diff
@@ -0,0 +1,123 @@
+import requests
+from configparser import ConfigParser
+from psycopg2 import connect
+from datetime import datetime, timedelta
+from tqdm import tqdm # this is a progress bar created by simply wrapping an iterable
+
+config = ConfigParser()
+config.read(r'/home/nwessel/db-creds.config')
+connection = connect(**config['dbauth'])
+connection.autocommit = True
+
+endpoint = 'https://apieco.eco-counter-tools.com'
+
+# get an authentication token for accessing the API
+def getToken():
+    config.read(r'volumes/ecocounter/.api-credentials.config')
+    response = requests.post(
+        f'{endpoint}/token',
+        headers={
+            'Authorization': 'Basic ' + config['DEFAULT']['secret_api_hash']
+        },
+        data={
+            'grant_type': 'password',
+            'username': config['DEFAULT']['username'],
+            'password': config['DEFAULT']['password']
+        }
+    )
+    return response.json()['access_token']
+
+# get a list of all sites from the API
+def getSites():
+    response = requests.get(
+        f'{endpoint}/api/site',
+        headers={'Authorization': f'Bearer {token}'}
+    )
+    return response.json()
+
+# gt all of a channel/flow's data from the API
+def getChannelData(channel_id,firstData='2015-01-01T00:00:00'):
+    requestChunkSize = timedelta(days=100)
+    requestStartDate = datetime.strptime(firstData,"%Y-%m-%dT%H:%M:%S%z")
+    data = []
+    while requestStartDate.timestamp() < datetime.now().timestamp():
+        response = requests.get(
+            f'{endpoint}/api/data/site/{channel_id}',
+            headers={'Authorization': f'Bearer {token}'},
+            params={
+                'begin': requestStartDate.isoformat(timespec='seconds'), 
```

Comment:
ideally the `begin` and `end` would be a param that we can call but this is great for meow

Reply:
Yeah, I mostly just wanted to demo for you how the basic API calls work and get something that would be functional for pulling all the data.